server/api/impl/admin/admin.go

```diff
@@ -6,17 +6,20 @@ package admin
 import (
 	pluginapi "github.com/mattermost/mattermost-plugin-api"
 	"github.com/mattermost/mattermost-plugin-apps/server/api"
+	"github.com/mattermost/mattermost-plugin-apps/server/api/impl/aws"
 )
 
 type Admin struct {
-	mm    *pluginapi.Client
-	conf  api.Configurator
-	store api.Store
-	proxy api.Proxy
+	mm         *pluginapi.Client
+	conf       api.Configurator
+	store      api.Store
+	proxy      api.Proxy
+	awsClient  *aws.Client
+	adminToken api.SessionToken // TODO populate admin token
```

Comment:
Is this the token referred to in https://mattermost.atlassian.net/browse/MM-28542 - used for upgrade ops only? If yes, should we call it adminToken or use some other name?

Reply:
Yes, that's the one. Agree, `adminToken` is kinda ambiguous name. In the similar sense `Admin` struct name is ambiguous too. This struct administers and manages the app installation, upgrade, etc. hence the name. It has nothing to do with the system admin and that's the misleading part of the name.
how about `administrationToken` or `proxyPluginToken` or `mainToken`. I'm 0/5

Reply:
3/5 Admin (access/session) token can not be a part of this struct - `Admin` is a "service, it lives for the entire duration of the proxy, across all apps and requests. The access token should be a request attribute.

Reply:
If we agree that all changes to apps are made only in the plugin's `onActivate` method, we can have short lived admin token scoped down to the `onActivate` - Create admin token, use for the synchronization and delete.
Another approach could be to create token once, save it in the `Admin` struct(and update it every month or so for security reasons?).
Anyway this effort is the part of the different ticket.